src/client.rs

```diff
@@ -56,6 +60,43 @@ pub(crate) type QueryFuture = Box<dyn Future<Output = Result<serde_json::Value>>
 #[cfg(any(target_arch = "wasm32", target_arch = "wasm64"))]
 pub(crate) type QueryFuture = Box<dyn Future<Output = Result<serde_json::Value>>>;
 
+/// Where to begin returning results from in paginated requests.
+#[derive(Debug, PartialEq, Eq, Clone, Copy, SerializeDisplay, DeserializeFromStr)]
+pub enum Cursor {
+    /// Begin at the given page. Actual offset depends on page size.
+    Page(u64),
+
+    /// Return page size items ordered before the given id.
+    Before(u64),
+
+    /// Return page size items ordered after the given id.
+    After(u64),
+}
+
+impl FromStr for Cursor {
+    type Err = ParseIntError;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        let result = match s.chars().next() {
+            Some('a') => Self::After(s[1..].parse()?),
+            Some('b') => Self::Before(s[1..].parse()?),
+            None | Some(_) => Self::Page(s.parse()?),
+        };
+
+        Ok(result)
+    }
+}
+
+impl fmt::Display for Cursor {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Page(p) => write!(f, "{}", p),
+            Self::Before(p) => write!(f, "b{}", p),
+            Self::After(p) => write!(f, "a{}", p),
```

Comment:
```suggestion
            Self::Page(p) => write!(f, "{p}"),
            Self::Before(p) => write!(f, "b{p}"),
            Self::After(p) => write!(f, "a{p}"),
```

Reply:
Should I also add `rust-version = "1.58"` to `Cargo.toml`?

Reply:
oh maybe, i didn't check what the msrv is

Reply:
Looks like it'd be around `1.70` because of some dependencies.

Reply:
yeah that seems okay to be honest?